lib/game.rb

```diff
@@ -0,0 +1,96 @@
+require_relative 'board'
+class Game
+  attr_reader :players, :board
+  attr_accessor :active_player
+
+  def initialize
+    @board = Board.new
+    @players = { 'Player 1' => 'X', 'Player 2' => 'O' }
+    @active_player = 'Player 1'
+  end
+
+  def add(column)
+    row = 5
+    while row >= 0
+      if @board.positions[row][column].nil?
+        @board.positions[row][column] = @players[@active_player]
+        return
+      end
+      row -= 1
+    end
+  end
+
+  def change_player
+    @active_player = if @active_player == 'Player 1'
+                       'Player 2'
+                     else
+                       'Player 1'
+                     end
+  end
+
+  def column_full?(column)
+    return false if @board.positions[0][column].nil?
+
+    true
+  end
+
+  def win?
+    player = @players[active_player]
+
+    # horizontal check
+    for j in 0..3 do
+      for i in 0..5 do
+        if @board.positions[i][j] == player && @board.positions[i][j + 1] == player &&
```

Comment:
For the very least, there is a need to have `Board#coin_at(column:, row:)`. Iteration over all these different ways to win can stay here, since it's a game logic.